nebula_tests/continental/continental_ros_decoder_test_srr520.cpp

```diff
@@ -0,0 +1,309 @@
+  drivers::continental_srr520::ContinentalSRR520SensorConfiguration & sensor_configuration)
+{
+  std::filesystem::path bag_root_dir =
+    _SRC_RESOURCES_DIR_PATH;  // variable defined in CMakeLists.txt;
+  bag_root_dir /= "continental";
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("sensor_model", "SRR520");
+    sensor_configuration.sensor_model =
+      nebula::drivers::SensorModelFromString(this->get_parameter("sensor_model").as_string());
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("base_frame", "some_base_frame", descriptor);
+    sensor_configuration.frame_id = this->get_parameter("base_frame").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("frame_id", "some_sensor_frame", descriptor);
+    sensor_configuration.base_frame = this->get_parameter("frame_id").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>(
+      "bag_path", (bag_root_dir / "srr520" / "1708578209").string(), descriptor);
+    bag_path_ = this->get_parameter("bag_path").as_string();
+    std::cout << bag_path_ << std::endl;
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("storage_id", "sqlite3", descriptor);
+    storage_id_ = this->get_parameter("storage_id").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>("format", "cdr", descriptor);
+    format_ = this->get_parameter("format").as_string();
+  }
+  {
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = 4;
+    descriptor.read_only = true;
+    descriptor.dynamic_typing = false;
+    descriptor.additional_constraints = "";
+    this->declare_parameter<std::string>(
+      "target_topic", "/sensing/radar/front_left/nebula_packets", descriptor);
+    target_topic_ = this->get_parameter("target_topic").as_string();
+  }
```

Comment:
Please also adopt the new syntax here.

Reply:
Would you be ok then having the parameter descriptor source file copied again?

Reply:
I'll refactor it to be header-only at some point but for now feel free to copy it.

Reply:
Addressed in 
[38771fb](https://github.com/tier4/nebula/pull/124/commits/38771fb343a40c816fd5c5adabd60ac506330a47)